File: capture_live_ppid.py
import os
import time
import subprocess
from PIL import Image

EDGE_PATH = r"C:\Program Files (x86)\Microsoft\Edge\Application\msedge.exe"
OUT_DIR = r"public\assets\03_Proyek_Website_PPID_PKTJ"
os.makedirs(OUT_DIR, exist_ok=True)

PAGES = [
    ("01_Beranda_Portal_PPID_PKTJ.jpg", "https://ppid.pktj.ac.id", "Beranda Portal Utama PPID PKTJ"),
    ("02_Profil_PPID.jpg", "https://ppid.pktj.ac.id/profil-ppid.html", "Profil PPID PKTJ"),
    ("03_Visi_dan_Misi.jpg", "https://ppid.pktj.ac.id/profil-visi-misi.html", "Visi dan Misi"),
    ("04_Struktur_Organisasi.jpg", "https://ppid.pktj.ac.id/profil-struktur-organisasi.html", "Struktur Organisasi"),
    ("05_Tugas_dan_Tanggung_Jawab.jpg", "https://ppid.pktj.ac.id/profil-tugas-tanggung-jawab.html", "Tugas dan Tanggung Jawab"),
    ("06_Regulasi_Acuan_PPID.jpg", "https://ppid.pktj.ac.id/profil-regulasi.html", "Regulasi Acuan PPID"),
    ("07_Kontak_PPID.jpg", "https://ppid.pktj.ac.id/profil-kontak.html", "Kontak Resmi PPID"),
    ("08_Layanan_Informasi_Menu.jpg", "https://ppid.pktj.ac.id/halaman/layanan-informasi-menu", "Menu Layanan Informasi"),
    ("09_Maklumat_Pelayanan.jpg", "https://ppid.pktj.ac.id/layanan-informasi/maklumat", "Maklumat Pelayanan Informasi"),
    ("10_Daftar_Informasi_Publik.jpg", "https://ppid.pktj.ac.id/layanan-informasi/daftar", "Daftar Informasi Publik (DIP)"),
    ("11_Laporan_Pelayanan_Informasi.jpg", "https://ppid.pktj.ac.id/layanan-informasi/laporan", "Laporan Pelayanan Informasi"),
    ("12_Laporan_Akses_Informasi.jpg", "https://ppid.pktj.ac.id/layanan-informasi/laporan-akses", "Laporan Akses Layanan Informasi"),
    ("13_Laporan_Survey_Kepuasan.jpg", "https://ppid.pktj.ac.id/layanan-informasi/laporan-survey", "Laporan Survey Kepuasan Masyarakat"),
    ("14_Informasi_Berkala.jpg", "https://ppid.pktj.ac.id/informasi-publik/berkala", "Informasi Publik Berkala"),
    ("15_Informasi_Serta_Merta.jpg", "https://ppid.pktj.ac.id/informasi-publik/serta-merta", "Informasi Publik Serta Merta"),
    ("16_Informasi_Setiap_Saat.jpg", "https://ppid.pktj.ac.id/informasi-publik/setiap-saat", "Informasi Publik Setiap Saat"),
    ("17_Informasi_Dikecualikan.jpg", "https://ppid.pktj.ac.id/informasi-publik/dikecualikan", "Informasi Publik Dikecualikan"),
    ("18_Permohonan_Informasi_Online.jpg", "https://ppid.pktj.ac.id/permohonan-informasi", "Formulir Permohonan Informasi Publik"),
    ("19_SOP_Permintaan_Informasi.jpg", "https://ppid.pktj.ac.id/prosedur/sop-permintaan", "SOP Prosedur Permintaan Informasi"),
    ("20_SOP_Pengajuan_Keberatan.jpg", "https://ppid.pktj.ac.id/prosedur/sop-keberatan", "SOP Prosedur Pengajuan Keberatan"),
    ("21_SOP_Penyelesaian_Sengketa.jpg", "https://ppid.pktj.ac.id/prosedur/sop-sengketa", "SOP Penyelesaian Sengketa Informasi"),
    ("22_FAQ_Pusat_Bantuan.jpg", "https://ppid.pktj.ac.id/faq", "FAQ / Pertanyaan Umum"),
    ("23_Berita_Sosialisasi_PPID.jpg", "https://ppid.pktj.ac.id/berita", "Portal Berita & Publikasi Kegiatan"),
    ("24_Formulir_Cetak_Permohonan.jpg", "https://ppid.pktj.ac.id/dokumen/formulir-permohonan-cetak", "Formulir Cetak Permohonan Fisik"),
]

print(f"Capturing live screenshots for {len(PAGES)} pages...")

captured_files = []
for filename, url, title in PAGES:
    out_file = os.path.join(OUT_DIR, filename)
    png_tmp = out_file.replace('.jpg', '_tmp.png')
    
    cmd = [
        EDGE_PATH,
        "--headless=new",
        "--disable-gpu",
        "--hide-scrollbars",
        "--window-size=1600,1000",
        f"--screenshot={png_tmp}",
        "--virtual-time-budget=4000",
        url
    ]
    
    print(f"Capturing: {title} ({url}) -> {filename} ...")
    try:
        subprocess.run(cmd, check=True, timeout=25, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
        if os.path.exists(png_tmp):
            img = Image.open(png_tmp)
            if img.mode != 'RGB':
                img = img.convert('RGB')
            # Save as high quality JPG
            img.save(out_file, 'JPEG', quality=92)
            os.remove(png_tmp)
            captured_files.append(out_file)
            print(f"  [OK] Saved {filename} ({os.path.getsize(out_file)} bytes)")
        else:
            print(f"  [WARN] Temporary PNG not generated for {url}")
    except Exception as e:
        print(f"  [ERROR] {e}")

print(f"\nTotal live screenshots captured: {len(captured_files)}")

# Now generate PDF
if captured_files:
    pdf_path = os.path.join(OUT_DIR, "Dokumentasi_Lengkap_Portal_PPID_PKTJ.pdf")
    pdf_images = [Image.open(f) for f in captured_files]
    pdf_images[0].save(pdf_path, save_all=True, append_images=pdf_images[1:], resolution=100.0, quality=90)
    print(f"\nGenerated live PDF documentation: {pdf_path} ({os.path.getsize(pdf_path)} bytes)")

File: capture_live_ppid.js
import puppeteer from 'puppeteer';
import path from 'path';
import fs from 'fs';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const OUT_DIR = path.join(__dirname, 'public', 'assets', '03_Proyek_Website_PPID_PKTJ');

if (!fs.existsSync(OUT_DIR)) {
  fs.mkdirSync(OUT_DIR, { recursive: true });
}

const PAGES = [
  { file: '01_Beranda_Portal_PPID_PKTJ.jpg', url: 'https://ppid.pktj.ac.id', title: '01. Beranda Portal PPID' },
  { file: '02_Profil_PPID.jpg', url: 'https://ppid.pktj.ac.id/profil-ppid.html', title: '02. Profil PPID PKTJ' },
  { file: '03_Visi_dan_Misi.jpg', url: 'https://ppid.pktj.ac.id/profil-visi-misi.html', title: '03. Visi dan Misi' },
  { file: '04_Struktur_Organisasi.jpg', url: 'https://ppid.pktj.ac.id/profil-struktur-organisasi.html', title: '04. Struktur Organisasi' },
  { file: '05_Tugas_dan_Tanggung_Jawab.jpg', url: 'https://ppid.pktj.ac.id/profil-tugas-tanggung-jawab.html', title: '05. Tugas dan Tanggung Jawab' },
  { file: '06_Regulasi_Acuan_PPID.jpg', url: 'https://ppid.pktj.ac.id/profil-regulasi.html', title: '06. Regulasi Acuan PPID' },
  { file: '07_Kontak_PPID.jpg', url: 'https://ppid.pktj.ac.id/profil-kontak.html', title: '07. Kontak Resmi PPID' },
  { file: '08_Layanan_Informasi_Menu.jpg', url: 'https://ppid.pktj.ac.id/halaman/layanan-informasi-menu', title: '08. Menu Layanan Informasi' },
  { file: '09_Maklumat_Pelayanan.jpg', url: 'https://ppid.pktj.ac.id/layanan-informasi/maklumat', title: '09. Maklumat Pelayanan' },
  { file: '10_Daftar_Informasi_Publik.jpg', url: 'https://ppid.pktj.ac.id/layanan-informasi/daftar', title: '10. Daftar Informasi Publik' },
  { file: '11_Laporan_Pelayanan_Informasi.jpg', url: 'https://ppid.pktj.ac.id/layanan-informasi/laporan', title: '11. Laporan Pelayanan Informasi' },
  { file: '12_Laporan_Akses_Informasi.jpg', url: 'https://ppid.pktj.ac.id/layanan-informasi/laporan-akses', title: '12. Laporan Akses Layanan' },
  { file: '13_Laporan_Survey_Kepuasan.jpg', url: 'https://ppid.pktj.ac.id/layanan-informasi/laporan-survey', title: '13. Laporan Survey Kepuasan' },
  { file: '14_Informasi_Berkala.jpg', url: 'https://ppid.pktj.ac.id/informasi-publik/berkala', title: '14. Informasi Publik Berkala' },
  { file: '15_Informasi_Serta_Merta.jpg', url: 'https://ppid.pktj.ac.id/informasi-publik/serta-merta', title: '15. Informasi Serta Merta' },
  { file: '16_Informasi_Setiap_Saat.jpg', url: 'https://ppid.pktj.ac.id/informasi-publik/setiap-saat', title: '16. Informasi Setiap Saat' },
  { file: '17_Informasi_Dikecualikan.jpg', url: 'https://ppid.pktj.ac.id/informasi-publik/dikecualikan', title: '17. Informasi Dikecualikan' },
  { file: '18_Permohonan_Informasi_Online.jpg', url: 'https://ppid.pktj.ac.id/permohonan-informasi', title: '18. Form Permohonan Informasi' },
  { file: '19_SOP_Permintaan_Informasi.jpg', url: 'https://ppid.pktj.ac.id/prosedur/sop-permintaan', title: '19. SOP Permintaan Informasi' },
  { file: '20_SOP_Pengajuan_Keberatan.jpg', url: 'https://ppid.pktj.ac.id/prosedur/sop-keberatan', title: '20. SOP Pengajuan Keberatan' },
  { file: '21_SOP_Penyelesaian_Sengketa.jpg', url: 'https://ppid.pktj.ac.id/prosedur/sop-sengketa', title: '21. SOP Penyelesaian Sengketa' },
  { file: '22_FAQ_Pusat_Bantuan.jpg', url: 'https://ppid.pktj.ac.id/faq', title: '22. FAQ & Pusat Bantuan' },
  { file: '23_Berita_Sosialisasi_PPID.jpg', url: 'https://ppid.pktj.ac.id/berita', title: '23. Portal Berita PPID' },
  { file: '24_Formulir_Cetak_Permohonan.jpg', url: 'https://ppid.pktj.ac.id/dokumen/formulir-permohonan-cetak', title: '24. Formulir Cetak Permohonan' },
];

async function run() {
  console.log('Launching headless browser...');
  const browser = await puppeteer.launch({
    headless: 'new',
    args: ['--no-sandbox', '--disable-setuid-sandbox', '--ignore-certificate-errors'],
  });

  const page = await browser.newPage();
  await page.setViewport({ width: 1440, height: 900 });

  const captured = [];

  for (let i = 0; i < PAGES.length; i++) {
    const item = PAGES[i];
    const targetPath = path.join(OUT_DIR, item.file);
    console.log(`[${i + 1}/${PAGES.length}] Capturing ${item.title} (${item.url})...`);

    try {
      await page.goto(item.url, { waitUntil: 'networkidle2', timeout: 30000 });
      // Small pause for any CSS animations to settle
      await new Promise((r) => setTimeout(r, 1200));

      await page.screenshot({
        path: targetPath,
        type: 'jpeg',
        quality: 90,
        fullPage: false,
      });

      console.log(`  -> Saved: ${item.file}`);
      captured.push(targetPath);
    } catch (err) {
      console.error(`  -> Failed ${item.url}:`, err.message);
    }
  }

  await browser.close();
  console.log(`\nSuccessfully captured ${captured.length} live pages!`);
}

run();
